### PR TITLE
Tidy up references to `gob`

### PR DESCRIPTION
**Pull request by tamird**
Minor progress toward #629.

### CODE REVIEW

File: gossip/gossip.go [hunk 1]
```diff
@@ -103,6 +103,13 @@ var (
 
 func init() {
 	gob.Register(&proto.NodeDescriptor{})
+	gob.Register(&proto.ZoneConfig{})
```
Comment (spencerkimball) [inline gossip/gossip.go]: Seems like it would be better to move these into util/gob.go (or similar). Strange to confine these to gossip.

**Comment by spencerkimball** [comment]
LGTM

File: gossip/gossip.go (hunk 1 above)
Comment (tamird) [inline gossip/gossip.go]: Isn't it the case that they're only useful as long as gossip uses the gob encoder/decoder?

Comment (spencerkimball) [inline gossip/gossip.go]: Ah, you're right. Yes, then this is the right place.

**Pull request merged**